### PR TITLE
Implement tests for "students/<int:student_id>/points" POST

### LINKED ISSUE #182: Implement tests for "students/<int:student_id>/points/" POST
**Endpoint description:** Add points to a student.

### PR DESCRIPTION
**Pull request by MartaSien**

### CODE REVIEW

**Review by arkjedrz — CHANGES_REQUESTED** [review]
Required changes were discussed yesterday.

File: tests/test_points.py [hunk 1]
```diff
@@ -84,3 +84,66 @@ def test_NoToken(self):
...
+    NOT_FOUND_URL = "/api/students/12345/points/"
+    # Fixture specific valid data to post a prize.
+    VALID_POST_PRIZE = {"content_type": "prize", "object_id": "2"}
+    # Fixture specific valid data to post a task.
+    VALID_POST_TASK = {"content_type": "task", "object_id": "2"}
+    # Fixture specific data to post an invalid content type.
+    INVALID_POST_TYPE = {"content_type": "asdf", "object_id": "1"}
+
+    def test_Prize_Success(self):
+        response = self.post(self.VALID_URL, self.VALID_POST_PRIZE)
```
Comment (arkjedrz) [inline tests/test_points.py:121]: I just noticed, that posting prize and task modifies student "total_points". This should also be checked.

File: tests/common.py [hunk 2]
```diff
@@ -168,3 +168,14 @@ def assert_not_found(self, response) -> None:
             response_json["detail"]
             == "You do not have permission to perform this action."
         )
+
+    def get_student_points(self, student_id: int) -> int:
```
Comment (arkjedrz) [inline tests/common.py]: Consider moving this method to `TestPointsPost` and renaming to `_get_student_points`.
I mean - this is an useful helper, but the usage is not really generic.

File: tests/test_points.py [hunk 3]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+
+    def test_Prize_Success(self):
+        student_points = self.get_student_points(2)
+        response = self.post(self.VALID_URL, self.VALID_POST_PRIZE)
+        # General assertions.
+        assert response.status_code == status.HTTP_201_CREATED
+        # Fixture specific assertions.
+        response_json = response.json()
+        student_prize = self.get("/api/students/2/prize/2/").json()
+        assert response_json["value"] == int(student_prize["value"])
```
Comment (arkjedrz) [inline tests/test_points.py:128]: There's no fault of the test here, but having to cast those values indicate some kind of mismatch between Task, Prize and Point implementations.
Why is "value" sometimes an int, and sometimes a string?

File: tests/test_points.py [hunk 4]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+        assert response_json["student"] == int(student_prize["student"])
+        assert response_json["object_id"] == student_prize["pk"]
+        assert response_json["assigner"] == 1
+        assert response_json["points_type"] == "prize"
+        assert response_json["content_type"] == 11
```
Comment (arkjedrz) [inline tests/test_points.py]: Consider creating a variable with a name that describes the content. It's not clear to me why it's expected that "content_type == 11".

File: tests/test_points.py [hunk 5]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+        assert response_json["object_id"] == student_task["pk"]
+        assert response_json["student"] == 2
+        assert response_json["assigner"] == 1
+        assert response_json["points_type"] == "task"
+        assert response_json["content_type"] == 10
+
+        points_added = self.get_student_points(2)
+        assert student_points + response_json["value"] == points_added
+
+    def test_Invalid_Type(self):
```
Comment (arkjedrz) [inline tests/test_points.py]: -> `test_InvalidType`

File: tests/test_points.py [hunk 6]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+    # Fixture specific valid data to post a task.
+    VALID_POST_TASK = {"content_type": "task", "object_id": "2"}
+    # Fixture specific data to post an invalid content type.
+    INVALID_POST_TYPE = {"content_type": "asdf", "object_id": "1"}
+
+    def test_Prize_Success(self):
+        student_points = self.get_student_points(2)
+        response = self.post(self.VALID_URL, self.VALID_POST_PRIZE)
+        # General assertions.
+        assert response.status_code == status.HTTP_201_CREATED
```
Comment (arkjedrz) [inline tests/test_points.py:123]: Add check for "application/json".

File: tests/test_points.py [hunk 7]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+        assert response_json["content_type"] == 11
+
+        points_spent = self.get_student_points(2)
+        assert student_points - response_json["value"] == points_spent
+
+    def test_Task_Success(self):
+        student_points = self.get_student_points(2)
+        response = self.post(self.VALID_URL, self.VALID_POST_TASK)
+        # General assertions.
+        assert response.status_code == status.HTTP_201_CREATED
```
Comment (arkjedrz) [inline tests/test_points.py:144]: Add check for "application/json".

File: tests/test_points.py [hunk 8]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+        response_json = response.json()
+        student_prize = self.get("/api/students/2/prize/2/").json()
+        assert response_json["value"] == int(student_prize["value"])
+        assert response_json["student"] == int(student_prize["student"])
+        assert response_json["object_id"] == student_prize["pk"]
+        assert response_json["assigner"] == 1
+        assert response_json["points_type"] == "prize"
+        assert response_json["content_type"] == 11
+
+        points_spent = self.get_student_points(2)
```
Comment (arkjedrz) [inline tests/test_points.py]: This is misleading - `get_student_points` return current number of points, not how many points were spent.

File: tests/test_points.py [hunk 9]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+        response_json = response.json()
+        student_task = self.get("/api/students/2/task/2/").json()
+        assert response_json["value"] == int(student_task["value"])
+        assert response_json["object_id"] == student_task["pk"]
+        assert response_json["student"] == 2
+        assert response_json["assigner"] == 1
+        assert response_json["points_type"] == "task"
+        assert response_json["content_type"] == 10
+
+        points_added = self.get_student_points(2)
```
Comment (arkjedrz) [inline tests/test_points.py]: This is misleading - `get_student_points` return current number of points, not how many points were added.

File: tests/test_points.py [hunk 10]
```diff
@@ -84,3 +84,78 @@ def test_NoToken(self):
...
+
+    def test_Task_Success(self):
+        student_points = self.get_student_points(2)
+        response = self.post(self.VALID_URL, self.VALID_POST_TASK)
+        # General assertions.
+        assert response.status_code == status.HTTP_201_CREATED
+        # Fixture specific assertions.
+        response_json = response.json()
+        student_task = self.get("/api/students/2/task/2/").json()
+        assert response_json["value"] == int(student_task["value"])
```
Comment (arkjedrz) [inline tests/test_points.py:149]: This is totally nitpick on my side :) but the order of checks is different than in `test_Prize_Success`.

**Review by arkjedrz — CHANGES_REQUESTED** [review]

File: tests/common.py (hunk 2 above)
Comment (MartaSien) [reply tests/common.py]: Done

File: tests/test_points.py (hunk 5 above)
Comment (MartaSien) [reply tests/test_points.py]: Done

File: tests/test_points.py (hunk 4 above)
Comment (MartaSien) [reply tests/test_points.py]: I proposed a new solution utilizing `ContentType`.

File: tests/test_points.py (hunk 8 above)
Comment (MartaSien) [reply tests/test_points.py]: Done

File: tests/test_points.py (hunk 9 above)
Comment (MartaSien) [reply tests/test_points.py]: Done

**Review by arkjedrz — APPROVED** [review]

**Comment by arkjedrz** [comment]
Consider using soft reset to squash 8 commits into one.

**Pull request merged**